### PR TITLE
feature(users-service): get user address by pin + shareable name

### PR DESCRIPTION
**Pull request by afkzoro**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
This PR creates the get user address by pin feature. 
#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

**Comment by afkzoro** [comment]
@siradji look into this.

File: apps/users-service/src/address-book-service.service.ts [hunk 1]
```diff
@@ -107,4 +113,33 @@ export class AddressBookService {
...
   }
+
+  async getAddressByPin (userId: string, pin: number): Promise<PinAddressI> {
+    try {
+      const getUser: UserI = await this.usersRepository.findOne({
+        _id: userId,
+        addressPin: pin
+      })
+
+      const addresses: AddressBookI[] = await this.repository.findAndPopulate(
```
Comment (siradji) [inline apps/users-service/src/address-book-service.service.ts:124]: Add a filter to only query sharable address. That is why we added the shareable: boolean field in the address schema

File: apps/api-gateway/src/module.api/address-book.controller.ts [hunk 2]
```diff
@@ -116,4 +117,25 @@ export class AddressBookController {
         )
     )
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Get('address/:pin')
+  async getAddressByPin (
+    @Param('pin') pin: number,
+      @CurrentUser() user: User
+  ): Promise<PinAddressI> {
```
Comment (siradji) [inline apps/api-gateway/src/module.api/address-book.controller.ts:126]: Do not pass the userId as a payload. Remember the person using this endpoint is not the user with this address right? The PIN will do

File: apps/users-service/src/address-book-service.service.ts [hunk 3]
```diff
@@ -107,4 +113,33 @@ export class AddressBookService {
 
     return { status: 1 }
   }
+
+  async getAddressByPin (userId: string, pin: number): Promise<PinAddressI> {
+    try {
+      const getUser: UserI = await this.usersRepository.findOne({
```
Comment (siradji) [inline apps/users-service/src/address-book-service.service.ts:119]: Only get the user with the PIN no userID. Remember the user hitting this endpoint is not the user with the address.

File: apps/users-service/src/address-book-service.service.ts [hunk 4]
```diff
@@ -107,4 +113,33 @@ export class AddressBookService {
...
+      const addresses: AddressBookI[] = await this.repository.findAndPopulate(
+        {
+          userId: getUser._id.toString(),
+          isDeleted: false
+        },
+        ['labelId']
+      )
+
+      return {
+        user: getUser,
```
Comment (siradji) [inline apps/users-service/src/address-book-service.service.ts:133]: No need to return the user in the PinAddressI. firstName, LastName, and address: AddressI (only adresses where sharable is true)

File: libs/common/src/database/dto/user/address.book.dto.ts [hunk 5]
```diff
@@ -19,4 +19,7 @@ export class AddressBookDto {
 
   @IsOptional()
     coordinates: [string, string]
+
```
Comment (siradji) [inline libs/common/src/database/dto/user/address.book.dto.ts:22]: add the sharable: boolean field

File: libs/common/src/typings/Global.Interface.ts [hunk 6]
```diff
@@ -354,3 +354,49 @@ export interface RegisterUserResponse {
   pinId: string
   phoneNumber: string
 }
+
+export interface PinAddressI {
+  user: UserI
```
Comment (siradji) [inline libs/common/src/typings/Global.Interface.ts:359]: remove the user Field

**Review by siradji — CHANGES_REQUESTED** [review]
@afkzoro, i left some comments

**Comment by afkzoro** [comment]
@siradji all fixed in latest commit.

File: apps/users-service/src/address-book-service.service.ts [hunk 7]
```diff
@@ -107,4 +113,32 @@ export class AddressBookService {
 
     return { status: 1 }
   }
+
+  async getAddressByPin (pin: number): Promise<PinAddressI> {
+    try {
+      const getUser: UserI = await this.usersRepository.findOne({
+        addressPin: pin
+      })
+
```
Comment (siradji) [inline apps/users-service/src/address-book-service.service.ts:122]: Handle cases where user with that pin can not be found. Throw a fit rpc error.

File: libs/common/src/database/dto/user/address.book.dto.ts [hunk 8]
```diff
@@ -19,4 +19,11 @@ export class AddressBookDto {
...
+  @IsOptional()
+    shareableName?: string
+
+  @IsNotEmpty()
+  @IsBoolean()
+    useOnce: boolean
```
Comment (siradji) [inline libs/common/src/database/dto/user/address.book.dto.ts:28]: What is useOnce?

**Review by siradji — CHANGES_REQUESTED** [review]

**Comment by afkzoro** [comment]
@siradji fixed.

**Comment by siradji** [comment]
Lgtm

**Comment by siradji** [comment]
Open a pr base is main against staging

**Pull request merged**